lib/WarnCaseSensitiveModulesPlugin.js

```diff
@@ -25,6 +26,17 @@ class WarnCaseSensitiveModulesPlugin {
 					const moduleWithoutCase = new Map();
 					for (const module of compilation.modules) {
 						const identifier = module.identifier();
+
+						// Ignore `data:` URLs, because it's not a real path
+						if (
+							/** @type {NormalModule} */
+							(module).resourceResolveData !== undefined &&
```

Comment:
What is the extra parenthesis around `(module)` for? Seems unnecessary.

Reply:
@TheLarkInn for typescirpt, because it is `Module` (base class), but we need `/** @type {NormalModule} */`